torch/csrc/jit/serialization/flatbuffer_serializer.cpp

```diff
@@ -0,0 +1,681 @@
+    offset = fbb.CreateStruct(mobile::serialization::Double(ivalue.toDouble()))
+                 .Union();
+  } else if (ivalue.isComplexDouble()) {
+    auto comp = ivalue.toComplexDouble();
+    ivalue_type = IValueUnion::ComplexDouble;
+    offset = fbb.CreateStruct(mobile::serialization::ComplexDouble(
+                                  comp.real(), comp.imag()))
+                 .Union();
+  } else if (ivalue.isInt()) {
+    ivalue_type = IValueUnion::Int;
+    offset =
+        fbb.CreateStruct(mobile::serialization::Int(ivalue.toInt())).Union();
+  } else if (ivalue.isBool()) {
+    ivalue_type = IValueUnion::Bool;
+    offset =
+        fbb.CreateStruct(mobile::serialization::Bool(ivalue.toBool())).Union();
+  } else if (ivalue.isString()) {
+    ivalue_type = IValueUnion::String;
+    offset = mobile::serialization::CreateString(
+                 fbb, fbb.CreateSharedString(ivalue.toString()->string()))
+                 .Union();
+  } else if (ivalue.isGenericDict()) {
+    ivalue_type = IValueUnion::Dict;
+    offset = dictToFB(fbb, ivalue).Union();
+  } else if (ivalue.isNone()) {
+    ivalue_type = IValueUnion::NONE;
+    offset = 0;
+  } else if (ivalue.isIntList()) {
+    ivalue_type = IValueUnion::IntList;
+    offset = mobile::serialization::CreateIntList(
+                 fbb, fbb.CreateVector(ivalue.toIntVector()))
+                 .Union();
+  } else if (ivalue.isDoubleList()) {
+    ivalue_type = IValueUnion::DoubleList;
+    offset = mobile::serialization::CreateDoubleList(
+                 fbb, fbb.CreateVector(ivalue.toDoubleVector()))
+                 .Union();
+  } else if (ivalue.isBoolList()) {
+    ivalue_type = IValueUnion::BoolList;
+    auto boollist = ivalue.toBoolList();
+    std::vector<uint8_t> bool_vec(boollist.begin(), boollist.end());
+    offset =
+        mobile::serialization::CreateBoolListDirect(fbb, &bool_vec).Union();
+  } else if (ivalue.isList()) {
+    ivalue_type = IValueUnion::List;
+    offset = listToFB(fbb, ivalue).Union();
+  } else if (ivalue.isObject()) {
+    ivalue_type = IValueUnion::Object;
+    offset = objectToFB(fbb, ivalue).Union();
+  } else if (ivalue.isDevice()) {
+    ivalue_type = IValueUnion::Device;
+    offset = mobile::serialization::CreateDevice(
+                 fbb, fbb.CreateSharedString(ivalue.toDevice().str()))
+                 .Union();
+  } else if (ivalue.isEnum()) {
+    const auto& enum_holder = ivalue.toEnumHolder();
+    const auto& qualified_class_name =
+        enum_holder->type()->qualifiedClassName();
+    uint32_t ival_pos = storeIValueAndGetIndex(fbb, enum_holder->value());
+    ivalue_type = IValueUnion::EnumValue;
+    offset = mobile::serialization::CreateEnumValue(
+                 fbb,
+                 fbb.CreateSharedString(qualified_class_name.qualifiedName()),
+                 ival_pos)
+                 .Union();
+  } else {
+    AT_ERROR("Unknown IValue type for pickling: ", ivalue.tagKind());
```

Comment:
LOL, I think we are done with "pickling"

Reply:
rephased